### PR TITLE
Game end & physics fix

### PR DESCRIPTION
**Pull request by marzsolt**
It seemed to work flawlessly.

### CODE REVIEW

File: src/Server/Game/Game.py [hunk 1]
```diff
@@ -25,26 +27,58 @@ def __init__(self):
...
+                # Check for deaths:
+                for pl in self.__player_logics:
+                    if pl.hp == 0:
+                        print("ID: ", pl._id, " Player has just died.")
+                        mess = BaseMessage(sermess.MessageType.DIED, sermess.Target.SCREEN)
+                        mess.player_id = pl._id
+                        Server.get_instance().send_all(mess)
+                        print("ID: ", pl._id, " All clients has been notified about the recent tragic death.")
+                        self.__player_logics = \
+                            [player_logic for player_logic in self.__player_logics if player_logic._id != pl._id]
```
Comment (DantElemer) [inline src/Server/Game/Game.py]: Same as before.

File: src/Server/Game/Game.py [hunk 2]
```diff
@@ -25,26 +27,58 @@ def __init__(self):
...
+                        print("ID: ", pl._id, " Player logic killed.")
+
+            # Check if there's a winner (last alive)
+            if len(self.__player_logics) == 1 and is_there_human:
+                mess = BaseMessage(sermess.MessageType.WON, sermess.Target.SCREEN)
+                mess.player_id = self.__player_logics[0]._id
+                Server.get_instance().send_all(mess)
+                Timer.sch_fun(1, self.stop_running, ())  # so that clients get the message
```
Comment (DantElemer) [inline src/Server/Game/Game.py]: See? It' a useful class! :)

File: src/Server/Game/Game.py [hunk 3]
```diff
@@ -25,26 +27,58 @@ def __init__(self):
...
-                        pl_i.hp -= 1 if pl_i.hp != 0 else 0
-                        pl_j.hp -= 1 if pl_j.hp != 0 else 0
+            else:
+                for i, pl_i in enumerate(self.__player_logics):
+                    # HP update for each player:
+                    for pl_j in self.__player_logics[i + 1:]:
+                        if (abs(pl_i.pos.x - pl_j.pos.x) <= 2 * PlayerLogic.RADIUS and
+                                abs(pl_i.pos.y - pl_j.pos.y) <= 2 * PlayerLogic.RADIUS):
+                            pl_i.hp -= 1 if pl_i.hp != 0 else 0
+                            pl_j.hp -= 1 if pl_j.hp != 0 else 0
```
[the remark this answers is not in the document]
Comment (marzsolt) [reply src/Server/Game/Game.py]: Nevertheless, I've changed to circle-like collision detection.

File: src/Client/Screen/Screen.py [hunk 4]
```diff
@@ -22,6 +22,10 @@ class Screen:
 
     BLACK = (0, 0, 0)
     WHITE = (255, 255, 255)
+    GREEN = (0, 255, 0)
+    RED = (255, 0, 0)
+
+    FPS = 40
```
[the remark this answers is not in the document]
Comment (marzsolt) [reply src/Client/Screen/Screen.py:29]: It is, indeed. Though, global constant file is yet to be created - for screen size, colors, fps, etc... .

File: src/Client/Player/PlayerManager.py [hunk 5]
```diff
@@ -46,4 +46,8 @@ def draw_players(self, screen):
         for p in self.players:
             screen.blit(p.surf, p.rect)
 
+    def remove_player(self, player_id):
+        self.players = [player for player in self.players if player._id != player_id]
```
[the remark this answers is not in the document]
Comment (marzsolt) [reply src/Client/Player/PlayerManager.py:50]: Well, I have just the player_id, thus I'd need a for cycle, and check there, then apply remove by value.

**Pull request merged**